src/pip/_internal/self_outdated_check.py

```diff
@@ -48,7 +48,7 @@ def __init__(self, cache_dir):
                 cache_dir, "selfcheck", _get_statefile_name(self.key)
             )
             try:
-                with open(self.statefile_path) as statefile:
+                with open(self.statefile_path, encoding="utf-8") as statefile:
```

Comment:
This is a backward compatibility break - at a minimum, we should check for encoding errors, and degrade gracefully if the existing file isn't UTF-8.

Reply:
This is JSON file encoded by UTF-8 (e.g. `six.ensure_binary()`). See [here](https://github.com/pypa/pip/blob/bbf8466088655d22cd46b286c8f0b8150754c1d9/src/pip/_internal/self_outdated_check.py#L85)

This was not a bug because data in the JSON is all ASCII, and `json.dumps()` is `ensure_ascii=True` by default.